### ISSUE TITLE
Please provide the ability to configure the number of nodes per hidden layer in networks.

### ISSUE THREAD

**Issue by Phong13** [Unity-Technologies/ml-agents#4622]
**Is your feature request related to a problem? Please describe.**
The config file for training only allows square networks (all layers have the same size e.g. 512x3 layers). In many white papers, the described networks have different numbers of nodes in each layer. For example:

features
[512]   hidden layer
[256]   hidden layer
[128]   hidden layer
output

I would like to be able to create networks with a different number of nodes per layer.  The reason for this is that the models should execute faster. In my case I have almost 500 observations but only 80 actions.  A network with fewer nodes in the deeper layers should infer almost as well but have few weights so it should execute faster.

**Comment by ervteng** [comment]
Agreed, this would be a useful feature. Hard part would be to make a simple entry in the configuration YAML that defines all of the layer configurations, without making it harder for beginner users. I'll log this request and we'll prioritize accordingly. 

Right now the fastest way to do it would be to edit the LinearEncoder in [layers.py](https://github.com/Unity-Technologies/ml-agents/blob/master/ml-agents/mlagents/trainers/torch/layers.py) and put in the new network architecture. It shouldn't have any issues exporting to ONNX, either.
